src/core/tranche/distributor/maker.sol

```diff
@@ -0,0 +1,92 @@
+// Copyright (C) 2019 Centrifuge
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+pragma solidity >=0.4.24;
+
+import "ds-note/note.sol";
+import "../distributor.sol";
+import "../../lightswitch.sol";
+
+contract MakerDistributor is Distributor, Switchable {
+
+    function balance() public auth {
+
+//        require(flowThrough==false);
+
+        if (manager.poolClosing()) {
+            repayTranches();
+        } else {
+            poolOpen();
+        }
+    }
+
+    function poolOpen() private {
+        int wad = manager.checkPile();
+        if (wad > 0) {
+            borrowFromTranches(uint(wad));
+        } else if (wad < 0) {
+            repayTranches();
+        }
+    }
+
+    function borrowFromTranches(uint wad) private {
+        uint borrowAmount = wad;
+        for (uint i = 0; i < manager.trancheCount(); i++) {
+            OperatorLike o = OperatorLike(manager.operatorOf(i));
+            uint trancheReserve = o.balance();
+            if (borrowAmount < trancheReserve) {
```

Comment:
```suggestion
            if (borrowAmount <= trancheReserve) {
```
I think needs to be like that. Otherwise considering the edge case:

`borrowAmount = 100, trancheReserve = 100`

The if-condition would be false, and we would borrow 100 from the first tranche.
The loop would continue and we would ask the next tranche to borrow `0`